src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/ContentCulturePickerController.cs

```diff
@@ -46,6 +53,18 @@ public async Task<IActionResult> RedirectToLocalizedContent(string targetCulture
             {
                 return LocalRedirect('~' + contentItemUrl);
             }
+            var setCookie = (await _siteService.GetSiteSettingsAsync()).As<ContentCulturePickerSettings>()?.SetCookie;
+
+            if(setCookie.HasValue && setCookie.Value)
+            {
+                // Set the cookie to handle redirecting to a custom controller
+                Response.Cookies.Delete(CookieRequestCultureProvider.DefaultCookieName);
+                Response.Cookies.Append(
+                    CookieRequestCultureProvider.DefaultCookieName,
+                    CookieRequestCultureProvider.MakeCookieValue(new RequestCulture(targetCulture)),
+                    new CookieOptions { Expires = DateTime.UtcNow.AddMonths(1) }
```

Comment:
Use what ASP.NET templates do for this cookie

Reply:
I assume that you mean the Authentication cookie ? 
https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authentication/Cookies/src/CookieAuthenticationOptions.cs#L31